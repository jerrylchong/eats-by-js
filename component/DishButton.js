import React from 'react';
import {StyleSheet, Text, View, TouchableOpacity, Dimensions} from 'react-native';

const DishButton = (props) => {

    const { title, description, price, onPress } = props;

    return (
        <TouchableOpacity style = {styles.container} onPress = {onPress}>
            <View style={styles.text}>
                <Text style={styles.name}>{title}</Text>
                <Text style={styles.cost}>${price}</Text>
            </View>
            <Text style = {styles.description}>{description}</Text>
        </TouchableOpacity>
    )
}

export default DishButton

const styles = StyleSheet.create({
    container: {
        paddingHorizontal:"3%",
        width: '100%',
        height: Dimensions.get('window').height * 0.10,
        alignSelf: 'center',
        justifyContent: 'center',
        borderBottomWidth: 0.5,
        borderColor: '#b3b3b3',
        backgroundColor: 'white'
    },
    text: {
        width: '100%',
        height: '30%',
        flexDirection: 'row',
        justifyContent: 'space-between',
        marginBottom: '5%',
        marginTop: '2%'
    },
    nameReview: {
        flexDirection: 'row',
        alignItems: 'center',
    },
    name: {
        color: 'black',
        fontSize: 20,
        fontFamily: 'Ubuntu-Bold'
    },
    cost: {
        color: 'black',
        fontSize: 20,
        fontFamily: 'Ubuntu'
    },
    description: {
        color: '#7e7e7e',
        fontSize: 11,
        alignSelf: 'flex-start',
        fontFamily: 'Ubuntu'
    },
})
