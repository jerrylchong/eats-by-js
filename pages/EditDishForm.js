import React, { useState } from 'react';
import {
    StyleSheet,
    TouchableOpacity,
    TextInput,
    KeyboardAvoidingView,
    Text,
    View,
    Platform,
    ImageBackground,
    Dimensions,
    TouchableWithoutFeedback,
    Keyboard,
    AsyncStorage
} from "react-native";
import { connect } from 'react-redux';
import {mapReduxStateToProps, mapReduxDispatchToProps} from "../helpers/reduxHelpers";
import {updateDish} from "../helpers/DishAPI";
import BackButton from "../component/BackButton";
import {useSafeArea} from "react-native-safe-area-context";

export function EditDishForm(props) {
    const {route, navigation} = props;
    const [errors, setErrors] = useState({});
    const [newTitle, setTitle] = useState('');
    const [newDescription, setDescription] = useState('');
    const [newPrice, setPrice] = useState('');
    const {id, title, desc, price} = route.params;

    const submit = () => {
        const attr =
            {
                title: newTitle ? newTitle : title,
                description: newDescription ? newDescription : desc,
                price: newPrice ? parseFloat(newPrice) : price
            };
        AsyncStorage.getItem('token').then(token => {
            updateDish(id, token, attr).then(() => {
                navigation.goBack();
            }).catch(err => {
                alert(err);
                console.log(err)
            })
        })
    }

    const backHandler = () => navigation.goBack();

    const insets = useSafeArea();

    return (
        <TouchableWithoutFeedback
            onPress={() => Keyboard.dismiss()}
        >
            <View
                style = {[
                    styles.container,
                    {paddingTop: insets.top, paddingLeft: insets.left, paddingRight: insets.right}
                ]}>
                <ImageBackground style = {styles.background} source={require('../assets/background.png')}/>
                <BackButton white={false} style = {{alignSelf: 'flex-start', margin: '2%'}} onPress = {() => navigation.goBack()} />
                <Text style = {styles.header}>Edit {title} dish</Text>
                <KeyboardAvoidingView behavior={Platform.OS == "ios" ? "padding" : "height"} style = {styles.list}>
                    <TextInput
                        style={styles.input}
                        placeholder={title}
                        onChangeText={(text) => {setTitle(text)}}
                        value={newTitle}
                        placeholderTextColor='#404040'/>
                    <TextInput
                        style={styles.input}
                        placeholder={desc}
                        onChangeText={(text) => {setDescription(text)}}
                        value={newDescription}
                        placeholderTextColor='#404040'/>
                    <TextInput
                        style={styles.input}
                        placeholder={price.toString()}
                        onChangeText={(text) => {setPrice(text)}}
                        value={newPrice}
                        placeholderTextColor='#404040'/>
                    { Object.entries(errors).map(x => x[0] + " " + x[1][0]).map((x,i) =>
                        <Text
                            key={`${i}-error`}
                            style={styles.error}>{x}</Text>
                    )}
                </KeyboardAvoidingView>
                <View style = {{position: 'relative', top: '60%', width: '100%', height: '15%', flexDirection:"row", justifyContent: 'space-evenly', alignItems: 'center'}}>
                    <View style = {styles.buttonShadow}>
                        <TouchableOpacity style = {styles.button} onPress = {submit}>
                            <Text style = {styles.buttonText}>Submit</Text>
                        </TouchableOpacity>
                    </View>
                </View>
            </View>
        </ TouchableWithoutFeedback>
    )
}

export default connect(mapReduxStateToProps,mapReduxDispatchToProps)(EditDishForm)

const windowWidth = Dimensions.get('window').width;

const styles = StyleSheet.create({
    background: {
        position:'absolute',
        bottom: 0,
        width: '100%',
        height: Dimensions.get('window').width * 728/1668,
    },
    container: {
        flex: 1,
        flexDirection: 'column',
        backgroundColor: '#fff',
        alignItems: 'center',
        minHeight: Math.round(Dimensions.get('window').height)
    },
    header: {
        fontSize: 20,
        fontFamily: 'Ubuntu-Medium',
        alignSelf: 'center'
    },
    list: {
        position: 'relative',
        top: '20%',
        width: '80%',
        height: '23%',
        alignItems: 'center',
        justifyContent: 'space-between'
    },
    inputHeader: {
        flexDirection: 'row',
        alignSelf: 'flex-start'
    },
    input: {
        borderRadius: windowWidth * 0.35,
        backgroundColor: '#d9d9d9',
        fontSize: 12,
        width: windowWidth * 0.8,
        height: windowWidth * 0.11,
        paddingHorizontal: '5%',
        fontFamily: 'Ubuntu',
        color: '#404040',
        fontWeight:'normal'
    },
    buttonShadow: {
        width: '25%',
        height: '30%',
        backgroundColor: '#ff6961',
        borderRadius: 20,
        shadowColor: "#000",
        shadowOffset: {
            width: 0,
            height: 2,
        },
        shadowOpacity: 0.25,
        shadowRadius: 3.84,
        elevation: 5,
    },
    button: {
        height: '100%',
        width: '100%',
        alignItems: 'center',
        justifyContent: 'center',
        paddingHorizontal: 20,
        paddingVertical: 3,
    },
    buttonText: {
        fontSize: 12,
        color: 'white',
        fontFamily: 'Ubuntu'
    },
    error: {
        fontSize: 14,
        color: '#fc8a1d',
        marginTop: '5%',
        fontFamily: 'Ubuntu'
    },
    top: {
        flexDirection: 'row',
        width: '100%',
    }
})
