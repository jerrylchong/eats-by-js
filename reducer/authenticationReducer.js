const authenticationReducer = (state = "not_set", action) => {
    switch (action.type) {
        case 'SET_TOKEN':
            return action.payload;
        case 'UNSET_TOKEN':
            return "not_set";
        default:
            return state;
    }
};
export default authenticationReducer
