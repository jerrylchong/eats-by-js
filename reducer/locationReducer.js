const locationReducer = (state = {hasLocation : false, coords : {}}, action) => {
    switch (action.type) {
        case 'SET_LOCATION':
            return {
                hasLocation : true,
                coords : {lat: action.payload.coords.latitude, lng: action.payload.coords.longitude}
            }
        case 'UNSET_LOCATION':
            return {
                hasLocation : false,
                coords : {lat: null, lng: null}
            }

        default:
            return state;
    }
};
export default locationReducer;
