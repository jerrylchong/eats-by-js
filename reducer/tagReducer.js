const tagReducer = (state = {tagData: [], isEmpty: true}, action) => {
    switch (action.type) {
        case 'SET_TAGS':
            return {
                tagData : action.payload,
                isEmpty: false
            }
        case 'UNSET_TAGS':
            return {
                tagData : [],
                isEmpty: true
            }

        default:
            return state;
    }
};
export default tagReducer;
