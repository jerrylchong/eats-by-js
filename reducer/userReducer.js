const userReducer = (state = {isLoggedIn: false, user_data:{}}, action) => {
    switch (action.type) {
        case 'SET_USER':
            return {
                isLoggedIn : true,
                user_data : action.payload
            }
        case 'UNSET_USER':
            return {
                isLoggedIn : false,
                user_data : {}
            }

        default:
            return state;
    }
};
export default userReducer;
