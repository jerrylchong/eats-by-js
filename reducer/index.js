import {combineReducers} from "redux";
import authenticationReducer from "./authenticationReducer"
import userReducer from "./userReducer";
import locationReducer from "./locationReducer";
import tagReducer from "./tagReducer";


export default combineReducers({
    token: authenticationReducer,
    user: userReducer,
    location: locationReducer,
    tags: tagReducer
})
