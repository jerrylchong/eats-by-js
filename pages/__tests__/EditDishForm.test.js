import React from 'react';
import renderer from 'react-test-renderer';
import Enzyme, {shallow} from "enzyme";
import Adapter from "enzyme-adapter-react-16";
import {createStore} from 'redux';
import {SafeAreaProvider} from "react-native-safe-area-context";
import reducer from '../../reducer'
import {Provider} from "react-redux";
import {EditDishForm} from "../EditDishForm";
import DishButton from "../../component/DishButton";

// snapshot test
test('renders correctly', () => {
    const store = createStore(reducer);
    const tree = renderer.create(
        <Provider store={store}>
            <SafeAreaProvider initialSafeAreaInsets={{ top: 0, left: 0, right: 0, bottom: 0 }}>
                <EditDishForm
                    navigation={{navigate: jest.fn()}}
                    route={{params: {title: 'Title', desc: 'Desc', price: '5'}}}

                />
            </SafeAreaProvider>
        </Provider>
    ).toJSON();
    expect(tree).toMatchSnapshot();
})

Enzyme.configure({ adapter: new Adapter() });