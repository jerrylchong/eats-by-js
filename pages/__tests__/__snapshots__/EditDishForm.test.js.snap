// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<RNCSafeAreaView
  onInsetsChange={[Function]}
  style={
    Object {
      "flex": 1,
    }
  }
>
  <View
    accessible={true}
    focusable={true}
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Array [
        Object {
          "alignItems": "center",
          "backgroundColor": "#fff",
          "flex": 1,
          "flexDirection": "column",
          "minHeight": 1334,
        },
        Object {
          "paddingLeft": 0,
          "paddingRight": 0,
          "paddingTop": 0,
        },
      ]
    }
  >
    <View
      accessibilityIgnoresInvertColors={true}
      style={
        Object {
          "bottom": 0,
          "height": 327.33812949640287,
          "position": "absolute",
          "width": "100%",
        }
      }
    >
      <Image
        source={1}
        style={
          Array [
            Object {
              "bottom": 0,
              "left": 0,
              "position": "absolute",
              "right": 0,
              "top": 0,
            },
            Object {
              "height": 327.33812949640287,
              "width": "100%",
            },
            undefined,
          ]
        }
      />
    </View>
    <View
      style={
        Object {
          "alignSelf": "flex-start",
          "margin": "2%",
        }
      }
    >
      <View
        accessible={true}
        focusable={true}
        isTVSelectable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          Object {
            "alignItems": "center",
            "height": 75,
            "justifyContent": "center",
            "opacity": 1,
            "width": 75,
          }
        }
      >
        <Image
          source={1}
          style={
            Object {
              "height": 13,
              "width": 15,
            }
          }
        />
      </View>
    </View>
    <Text
      style={
        Object {
          "alignSelf": "center",
          "fontFamily": "Ubuntu-Medium",
          "fontSize": 20,
        }
      }
    >
      Add a Dish
    </Text>
    <View
      onLayout={[Function]}
      style={
        Array [
          Object {
            "alignItems": "center",
            "height": "23%",
            "justifyContent": "space-between",
            "position": "relative",
            "top": "20%",
            "width": "80%",
          },
          Object {
            "paddingBottom": 0,
          },
        ]
      }
    >
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Title"
        placeholderTextColor="#404040"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "borderRadius": 262.5,
            "color": "#404040",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontWeight": "normal",
            "height": 82.5,
            "paddingHorizontal": "5%",
            "width": 600,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="Desc"
        placeholderTextColor="#404040"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "borderRadius": 262.5,
            "color": "#404040",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontWeight": "normal",
            "height": 82.5,
            "paddingHorizontal": "5%",
            "width": 600,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
      <TextInput
        allowFontScaling={true}
        onChangeText={[Function]}
        placeholder="5"
        placeholderTextColor="#404040"
        rejectResponderTermination={true}
        style={
          Object {
            "backgroundColor": "#d9d9d9",
            "borderRadius": 262.5,
            "color": "#404040",
            "fontFamily": "Ubuntu",
            "fontSize": 12,
            "fontWeight": "normal",
            "height": 82.5,
            "paddingHorizontal": "5%",
            "width": 600,
          }
        }
        underlineColorAndroid="transparent"
        value=""
      />
    </View>
    <View
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": "15%",
          "justifyContent": "space-evenly",
          "position": "relative",
          "top": "60%",
          "width": "100%",
        }
      }
    >
      <View
        style={
          Object {
            "backgroundColor": "#ff6961",
            "borderRadius": 20,
            "elevation": 5,
            "height": "30%",
            "shadowColor": "#000",
            "shadowOffset": Object {
              "height": 2,
              "width": 0,
            },
            "shadowOpacity": 0.25,
            "shadowRadius": 3.84,
            "width": "25%",
          }
        }
      >
        <View
          accessible={true}
          focusable={true}
          isTVSelectable={true}
          onClick={[Function]}
          onResponderGrant={[Function]}
          onResponderMove={[Function]}
          onResponderRelease={[Function]}
          onResponderTerminate={[Function]}
          onResponderTerminationRequest={[Function]}
          onStartShouldSetResponder={[Function]}
          style={
            Object {
              "alignItems": "center",
              "height": "100%",
              "justifyContent": "center",
              "opacity": 1,
              "paddingHorizontal": 20,
              "paddingVertical": 3,
              "width": "100%",
            }
          }
        >
          <Text
            style={
              Object {
                "color": "white",
                "fontFamily": "Ubuntu",
                "fontSize": 12,
              }
            }
          >
            Submit
          </Text>
        </View>
      </View>
    </View>
  </View>
</RNCSafeAreaView>
`;
